spring-cloud-stream/src/main/java/org/springframework/cloud/stream/aggregate/ModuleAggregationUtils.java

```diff
@@ -0,0 +1,154 @@
+		LinkedHashMap<String, ConfigurableApplicationContext> children = createChildContexts(parentContext, modules, args);
+		for (Map.Entry<String, ConfigurableApplicationContext> childContextEntry : children.entrySet()) {
+			ConfigurableApplicationContext childContext = childContextEntry.getValue();
+			parentContext.getBeanFactory().registerSingleton(childContextEntry.getKey(),
+					new BindableContextWrapper(childContext));
+			childContext.setParent(parentContext);
+		}
+	}
+
+	private static ConfigurableApplicationContext createParentContext(String[] args) {
+		SpringApplicationBuilder aggregatorParentConfiguration
+				= new SpringApplicationBuilder();
+		aggregatorParentConfiguration.sources(AggregatorParentConfiguration.class)
+				.web(false)
+				.headless(true)
+				.properties("spring.jmx.default-domain=" + UUID.randomUUID().toString().replace("-", ""))
+				.listeners(new UnbindOnCloseApplicationListener());
+		return aggregatorParentConfiguration.run(args);
+	}
+
+	private static LinkedHashMap<String, ConfigurableApplicationContext> createChildContexts(ConfigurableApplicationContext applicationContext, Class<?>[] modules,
+																																										 String args[][]) {
+		LinkedHashMap<String, ConfigurableApplicationContext> children = new LinkedHashMap<>();
+		for (int i = modules.length - 1; i >= 0; i--) {
+			String moduleClassName = modules[i].getName();
+			SpringApplicationBuilder childModuleBuilder = embedModule(applicationContext, getNamespace(moduleClassName, i), modules[i]);
+			ConfigurableApplicationContext childContext = childModuleBuilder.run(args != null ? args[i] : new String[0]);
+			children.put(getNamespace(moduleClassName,i), childContext);
+		}
+		return children;
+	}
+
+	private static String getNamespace(String moduleClassName, int index) {
+		return moduleClassName + "_" + index;
+	}
+
+	private static SpringApplicationBuilder embedModule(ConfigurableApplicationContext applicationContext, String namespace,
+																										 Class<?> module) {
+		return new SpringApplicationBuilder(module)
+				.web(false)
+				.showBanner(false)
+				.properties(BindableProxyFactory.CHANNEL_NAMESPACE_PROPERTY_NAME + "=" + namespace)
+				.registerShutdownHook(false)
+				.parent(applicationContext);
+	}
+
+	private static void prepareSharedChannelRegistry(SharedChannelRegistry sharedChannelRegistry, Class<?>[] modules) {
+		DirectChannel sharedChannel = null;
+		for (int i = 0; i < modules.length; i++) {
+			Class<?> module = modules[i];
+			String moduleClassName = module.getName();
+			if (i > 0) {
+				sharedChannelRegistry.putSharedChannel(getNamespace(moduleClassName,i) + ".input", sharedChannel);
+			}
+			sharedChannel = new DirectChannel();
+			if (i < modules.length - 1) {
+				sharedChannelRegistry.putSharedChannel(getNamespace(moduleClassName,i) + ".output", sharedChannel);
+			}
+		}
+	}
+
+	/**
+	 * Basic configuration for a parent
+	 */
+	@EnableAutoConfiguration
+	@EnableModule
```

Comment:
Related to the discussion we had [before] (https://github.com/mbogoevici/spring-cloud-stream/commit/f6bc3f1ae156f60f08cfb4cc80e76e8dac76d925#commitcomment-12935029), having `@EnableModule` would also make the `ChannelBindingLifecycle` bean to be part of the aggregator parentContext. This also means, every time the parent and child contexts (for the individual modules) are refreshed, we would have this bean triggered. This is harmless as of now because:
1) Parent context will not have any `bindable` beans
2) Each child context will have it's own bindable beans

I still think we can avoid this behavior of starting the binding lifecycle from the parent.